src/NixpkgsReview.hs

```diff
@@ -0,0 +1,48 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module NixpkgsReview
+  ( cacheDir,
+    runReport,
+  )
+where
+
+import Data.Text as T
+import qualified File as F
+import OurPrelude
+import qualified Process as P
+import System.Environment.XDG.BaseDir (getUserCacheDir)
+import Prelude hiding (log)
+
+cacheDir :: IO FilePath
+cacheDir = getUserCacheDir "nixpkgs-review"
+
+revDir :: FilePath -> Text -> FilePath
+revDir cache commit = cache <> "/rev-" <> T.unpack commit
+
+run ::
+  Members '[F.File, P.Process] r =>
+  FilePath ->
+  Text ->
+  Sem r Text
+run cache commit = do
+  -- TODO: probably just skip running nixpkgs-review if the directory
+  -- already exists
```

Comment:
One thing to note is that `nixpkgs-review` will do repeat reviews in new directories by suffixing `-1`, `-2`, `-3`, ... to the directory name where it outputs, but I guess the `rm -rf` handles it to make sure we always go to the unsufixed dir and also never get stale data?

Reply:
Right that's what `rm -rf` is for.